test/test_torch.py

```diff
@@ -10878,6 +10878,27 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*(torch.testing.get_all_fp_dtypes(include_half=False) +
```

Comment:
I'm surprised you don't have to put @dtypes before @dtypesIfCUDA. Even if it's not required it would make more logical sense.

Reply:
What do you mean? These lines have `@dtypes` before `@dtypesIfCUDA`.

Reply:
OK I got what you meant. This order is used everywhere else AFAIK. I would keep it consistent here